Refactoring.Pipelines/JoinedPipes.cs

```diff
@@ -21,8 +22,8 @@ public class JoinedPipes<TInput1, TInput2> : Sender<Tuple<TInput1, TInput2>>, IG
         private readonly ForwardingListener<TInput2> _listener2;
         private readonly ISender<TInput1> _sender1;
         private readonly ISender<TInput2> _sender2;
-        private readonly Queue<TInput1> _values1 = new Queue<TInput1>();
-        private readonly Queue<TInput2> _values2 = new Queue<TInput2>();
+        private readonly ConcurrentQueue<TInput1> _values1 = new ConcurrentQueue<TInput1>();
+        private readonly ConcurrentQueue<TInput2> _values2 = new ConcurrentQueue<TInput2>();
```

Comment:
Not sure if I need to subclass and create a new JoinedPipes in the Async namespace with just the ConcurrentQueue part.